controllers/grid/navigationMenus/form/NavigationMenuForm.inc.php

```diff
@@ -207,6 +206,13 @@ function validate() {
 			$this->addError('path', __('manager.navigationMenus.form.duplicateTitle'));
 		}
 
+		$navigationMenu = $navigationMenuDao->getById($this->_navigationMenuId);
+
+		$navigationMenusWithArea = $navigationMenuDao->getAllByArea($this->_contextId, $this->getData('areaName'))->toArray();
+		if (count($navigationMenusWithArea) == 1 && $this->getData('areaName') != '' && $navigationMenusWithArea[0]->getId() != $this->_navigationMenuId) {
+			$this->addError('areaName', __('manager.navigationMenus.form.lastOfArea'));
+		}
+
```

Comment:
Check `$this->getData('areaName')` before making the DAO call so we can save a DB request if possible:

```
if ($this->getData('areaName') !== '') {
	$assignedMenu = $navigationMenuDao->getByArea($this->_contextId, $this->getData('areaName'));
	if ($assignedMenu && $assignedMenu->getId() !== $navigationMenu->getId()) {
		$this->addError('areaName', __('manager.navigationMenus.form.assignMultiple'));
	}
}
```

Reply:
OK